packages/dx-react-grid-bootstrap3/src/templates/table-group-cell/container.jsx

```diff
@@ -0,0 +1,35 @@
+import React, { useState, useContext } from 'react';
+import * as PropTypes from 'prop-types';
+import { getStickyPosition } from '../../utils/css-fallback-properties';
+import { ThemeColors } from '../layout';
+
+export const Container = ({ children, style, ...restProps }) => {
+  const [position] = useState(getStickyPosition());
```

Comment:
With `setState` result of `useState` omitted, `useState` is the same as just assignment, isn't it?

```jsx
const [y] = useState(x);
// or
const y = x;
```

Shouldn't it be something like that?
```jsx
const [position, setPosition] = useState(getStickyPosition());
useEffect(() => {
  setPosition(getStickyPosition());
}, []);
```

Reply:
> With setState result of useState omitted, useState is the same as just assignment, isn't it?

Yes, this is intended - we need to initialize this variable only once on component render

Reply:
Do you mean that `getStickyPosition` must be called only once in a component lifetime (like in `constructor`)?

Reply:
Yes, exactly - like in table fixed cell component

Reply:
It is rather strange - `getStickyPosition` will be called on each component update (without purpose) though the result of the first call will always be used.
https://codesandbox.io/s/billowing-glitter-ij0my

As for `useEffect` - it seems that it can be omitted. And `componentDidMount` in `FixedCell` can be omitted as well.

----
But actually - both should be kept.
`getStickyPosition` is not pure (it accesses DOM) - hence it should not be called neither from `render` nor from `constructor`.
Sticky position usage seems to be like this - state field is initialized with `undefined` (or something like this, but not with `getStickyPosition()`) and on *mount* stage field is updated with `getStickyPosition()`. Doesn't it?

Reply:
We better move it to a context and calculate it once per page load - now a sticky position gets calculated once per row at best